src/shared/hooks/useIntersectionObserver.test.tsx

```diff
@@ -0,0 +1,109 @@
+import { render } from '@testing-library/react';
+
+import { useIntersectionObserver } from './useIntersectionObserver';
+
+const instanceMap = new Map();
+const observerMap = new Map();
+
+export function intersect(element: Element, isIntersecting: boolean) {
+  const cb = observerMap.get(element);
+
+  if (cb) {
+    cb([
+      {
+        isIntersecting,
+        target: element,
+        intersectionRatio: isIntersecting ? 1 : -1,
+      },
+    ]);
+  }
+}
+
+export function getObserverOf(element: Element): IntersectionObserver {
+  return instanceMap.get(element);
+}
+
+const Observed = ({ callback = () => {} }: { callback?: () => void }) => {
+  useIntersectionObserver({
+    targetSelector: '.observed',
+    loadNextPage: callback,
+  });
+
+  return (
+    <div data-testid="wrapper">
+      <div className="observed" data-testid="observed">
+        {' '}
+      </div>
+    </div>
+  );
+};
+
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+beforeEach(() => {
```

Comment:
why did you place `beforeEach` and `afterEach` outside of parent `describe`?

Reply:
fixed